buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java

```diff
@@ -0,0 +1,378 @@
+            test.getExtensions().add("errorReportingTestListener", listener);
+            test.addTestOutputListener(listener);
+            test.addTestListener(listener);
+
+            /*
+             * We use lazy-evaluated strings in order to configure system properties whose value will not be known until
+             * execution time (e.g. cluster port numbers). Adding these via the normal DSL doesn't work as these get treated
+             * as task inputs and therefore Gradle attempts to snapshot them before/after task execution. This fails due
+             * to the GStrings containing references to non-serializable objects.
+             *
+             * We bypass this by instead passing this system properties vi a CommandLineArgumentProvider. This has the added
+             * side-effect that these properties are NOT treated as inputs, therefore they don't influence things like the
+             * build cache key or up to date checking.
+             */
+            SystemPropertyCommandLineArgumentProvider nonInputProperties = new SystemPropertyCommandLineArgumentProvider();
+
+            test.doFirst(t -> {
+                project.mkdir(testOutputDir);
+                project.mkdir(heapdumpDir);
+                project.mkdir(test.getWorkingDir());
+                project.mkdir(test.getWorkingDir().toPath().resolve("temp"));
+
+                // TODO remove once jvm.options are added to test system properties
+                test.systemProperty("java.locale.providers", "SPI,COMPAT");
+            });
+            if (inFipsJvm()) {
+                project.getDependencies().add("testRuntimeOnly", "org.bouncycastle:bc-fips:1.0.1");
+                project.getDependencies().add("testRuntimeOnly", "org.bouncycastle:bctls-fips:1.0.9");
+            }
+            test.getJvmArgumentProviders().add(nonInputProperties);
+            test.getExtensions().add("nonInputProperties", nonInputProperties);
+
+            test.setWorkingDir(project.file(project.getBuildDir() + "/testrun/" + test.getName()));
+            test.setMaxParallelForks(Integer.parseInt(System.getProperty("tests.jvms", BuildParams.getDefaultParallel().toString())));
+
+            test.exclude("**/*$*.class");
+
+            test.jvmArgs(
+                "-Xmx" + System.getProperty("tests.heap.size", "512m"),
+                "-Xms" + System.getProperty("tests.heap.size", "512m"),
+                "--illegal-access=warn",
+                "-XX:+HeapDumpOnOutOfMemoryError"
+            );
+
+            test.getJvmArgumentProviders().add(() -> List.of("-XX:HeapDumpPath=$heapdumpDir"));
+
+            String argline = System.getProperty("tests.jvm.argline");
+            if (argline != null) {
+                test.jvmArgs((Object[]) argline.split(" "));
+            }
+
+            if (Boolean.parseBoolean(System.getProperty("tests.asserts", "true"))) {
```

Comment:
We should probably start replacing these with `Util.getBooleanProperty` for consistent bahavior.